src/SimpleAuth/OAuth/OAuthApi.cs

```diff
@@ -134,7 +135,8 @@ public override void ResetData()
 				{
 					if (!(await Ping(TokenUrl)))
 						return account;
-					await RefreshAccount(account);
+					if (await RefreshAccount(account))
+						account = CurrentOAuthAccount ?? GetAccount<OAuthAccount>(Identifier);
```

Comment:
This doesn't seem right. After we refresh the account you are potentially overwriting it.

Especially since the next line is verify that the account is correct. And if so, save it and set it to the CurrentAccount